desc/objectives/_coils.py

```diff
@@ -98,77 +98,48 @@ def build(self, use_jit=True, verbose=1):  # noqa:C901
         # local import to avoid circular import
         from desc.coils import CoilSet, MixedCoilSet
 
+        coil = self.things[0]
+
         self._dim_f = 0
         self._quad_weights = jnp.array([])
 
-        def to_list(coilset):
-            """Turn a MixedCoilSet container into a list of what it's containing."""
-            if isinstance(coilset, list):
-                return [to_list(x) for x in coilset]
-            elif isinstance(coilset, MixedCoilSet):
-                return [to_list(x) for x in coilset]
-            elif isinstance(coilset, CoilSet):
-                # use the same grid/transform for CoilSet
-                return to_list(coilset.coils[0])
-            else:
-                return [coilset]
-
-        # gives structure of coils, e.g. MixedCoilSet(coils, coils) would give a
-        # a structure of [[*, *], [*, *]] if n = 2 coils
-        coil_leaves, coil_structure = tree_flatten(
-            self.things[0], is_leaf=lambda x: not hasattr(x, "__len__")
-        )
-        self._num_coils = len(coil_leaves)
+        # get individual coils from coilset
+        coils = tree_leaves(coil, is_leaf=lambda x: hasattr(x, "_current"))
```

Comment:
I'd prefer something like 

```python
def _is_single_coil(x):
    return isinstance(x, _Coil) and not isinstance(x, CoilSet)

...
is_leaf=_is_single_coil
```

rather than checking if it has `_current`, since other objects may have a `_current` attribute (like equilibrium, winding surface, etc).